gotools.mk

```diff
@@ -27,31 +27,25 @@ gotools-clean:
 # Special override for protoc-gen-go since we want to use the version vendored with the project
 gotool.protoc-gen-go:
 	@echo "Building github.com/golang/protobuf/protoc-gen-go -> protoc-gen-go"
-	GOBIN=$(abspath $(GOTOOLS_BINDIR)) go install ./vendor/github.com/golang/protobuf/protoc-gen-go
+	GOBIN=$(abspath $(GOTOOLS_BINDIR)) go get github.com/golang/protobuf/protoc-gen-go@v1.3.1
 
 # Special override for ginkgo since we want to use the version vendored with the project
 gotool.ginkgo:
 	@echo "Building github.com/onsi/ginkgo/ginkgo -> ginkgo"
-	GOBIN=$(abspath $(GOTOOLS_BINDIR)) go install ./vendor/github.com/onsi/ginkgo/ginkgo
+	GOBIN=$(abspath $(GOTOOLS_BINDIR)) go get -v github.com/onsi/ginkgo/ginkgo@v1.8.0
 
 # Special override for goimports since we want to use the version vendored with the project
 gotool.goimports:
 	@echo "Building golang.org/x/tools/cmd/goimports -> goimports"
-	GOBIN=$(abspath $(GOTOOLS_BINDIR)) go install ./vendor/golang.org/x/tools/cmd/goimports
+	GOBIN=$(abspath $(GOTOOLS_BINDIR)) go get golang.org/x/tools/cmd/goimports@release-branch.go1.13
 
 # Special override for golint since we want to use the version vendored with the project
 gotool.golint:
 	@echo "Building golang.org/x/lint/golint -> golint"
-	GOBIN=$(abspath $(GOTOOLS_BINDIR)) go install ./vendor/golang.org/x/lint/golint
+	GOBIN=$(abspath $(GOTOOLS_BINDIR)) go get golang.org/x/lint/golint@959b441ac422379a43da2230f62be024250818b0
 
-# Lock to a versioned dep
-gotool.dep: DEP_VERSION ?= "v0.5.3"
-gotool.dep:
-	@GOPATH=$(abspath $(GOTOOLS_GOPATH)) go get -d -u github.com/golang/dep
-	@git -C $(abspath $(GOTOOLS_GOPATH))/src/github.com/golang/dep checkout -q $(DEP_VERSION)
-	@echo "Building github.com/golang/dep $(DEP_VERSION) -> dep"
-	@GOPATH=$(abspath $(GOTOOLS_GOPATH)) GOBIN=$(abspath $(GOTOOLS_BINDIR)) go install -ldflags="-X main.version=$(DEP_VERSION) -X main.buildDate=$$(date '+%Y-%m-%d')" github.com/golang/dep/cmd/dep
-	@git -C $(abspath $(GOTOOLS_GOPATH))/src/github.com/golang/dep checkout -q master
+gotool.counterfeiter:
+	GOPATH=$(abspath $(GOTOOLS_GOPATH)) GOBIN=$(abspath $(GOTOOLS_BINDIR)) GO111MODULE=off go get -u github.com/maxbrunsfeld/counterfeiter
 
 # Default rule for gotools uses the name->path map for a generic 'go get' style build
 gotool.%:
```

Comment:
Getting the floating version here can be locked down by doing a tools.go

See the PR note for more info. I should move that information into the commit message.